src/ui/molecules/select/Select.tsx

```diff
@@ -0,0 +1,317 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import type { RefObject } from 'react'
+import classNames from 'classnames'
+import {
+  capitalizeFirstLetter,
+  capitalizeFirstLetterOfEachArrayWord,
+  compareStrings,
+  isString
+} from 'utils'
+import type { UseState } from 'superTypes'
+import { InputBase } from 'ui/atoms/inputBase/InputBase'
+import type { InputBaseProps } from 'ui/atoms/inputBase/InputBase'
+import { Typography } from 'ui/atoms/typography/Typography'
+import { Icon } from 'ui/atoms/icon/Icon'
+import short from 'short-uuid'
+import './select.scss'
+
+export type Option = {
+  readonly label: string
+  readonly value: string
+  readonly group?: string
+}
+
+type InputPropsForSelect = Pick<InputBaseProps, 'placeholder' | 'disabled' | 'hasError'>
+
+export type SelectProps = InputPropsForSelect & {
+  /**
+   * Defines the callback called when the select value changes.
+   */
+  readonly onChange: (value: string | Readonly<string[]>) => void
+  /**
+   * The value of the select.
+   */
+  readonly value: string | string[]
```

Comment:
Consider renaming to something like `selected` to make it more clear what the value is

Reply:
In general the value selected by the select is called `value`, I prefer to keep this usage.